### ISSUE TITLE
Incomplete description of LSTM.__call__() params

### ISSUE THREAD

**Issue by habernal** [tensorflow/tensorflow#31237]
## URL(s) with the issue:

https://github.com/tensorflow/tensorflow/blob/48af54a586790cc29150be3d8665d7e8a1770257/tensorflow/python/keras/layers/recurrent.py#L2461

## Description of issue (what needs changing):

The current documentation reads:

```
  Call arguments:
    inputs: A 3D tensor.
    mask: Binary tensor of shape `(samples, timesteps)` indicating whether
      a given timestep should be masked.
    training: Python boolean indicating whether the layer should behave in
      training mode or in inference mode. This argument is passed to the cell
      when calling it. This is only relevant if `dropout` or
      `recurrent_dropout` is used.
    initial_state: List of initial state tensors to be passed to the first
      call of the cell.
```

It would be worth mentioning that

- mask, training, and initial_state are optional
- if initial_state is not provided, default zeros are imputed (by calling `cell.get_initial_state()` or `cell.zero_state()`), see tensorflow_core/python/ops/rnn.py lines 677 or 1382.

Due to iheritance of RNN layers (incl. mixins) and missing documentation at some places (haven't found any comments on `get_initial_state`), it's quite hard to figure it out.

### Submit a pull request?

I'd like to keep it to someone who really knows the Keras internals; to me it's still a bit cryptic (i.e., where exactly `get_initial_state_fn` comes from, etc.)

**Comment by lamberta** [comment]
Thanks. That file [is here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/layers/recurrent.py), can you send a pull request?

**Comment by habernal** [comment]
@lamberta I'm not sure what you mean

**Comment by habernal** [comment]
Fixed in this PR: https://github.com/tensorflow/tensorflow/pull/31646
